app/helpers/provisioning_templates_helper.rb

```diff
@@ -67,6 +67,15 @@ def permitted_actions(template)
     end
   end
 
+  def pxe_with_building_hosts?(template)
+    kinds = ["PXELinux", "PXEGrub"]
+    os_ids = template.operatingsystem_ids
+    template.respond_to?(:template_kind) &&
+      !template.template_kind.nil? &&
```

Comment:
Nitpick, `template.template_kind.present?` rather than the negation, or even combine it with the line above and do `template.try(:template_kind).present?` which'll work when it doesn't respond to.